models/contacts.js

```diff
@@ -1,19 +1,81 @@
-// const fs = require('fs/promises')
+const fs = require('fs').promises;
+const path = require('path');
+const contactsPath = path.join(__dirname, './contacts.json');
 
-const listContacts = async () => {}
+const listContacts = async () => {
+  try {
+    const data = await fs.readFile(contactsPath)
+    const contacts = JSON.parse(data);
+    return contacts;
+  } catch (err) {
+    console.log(err.message);
+  }
+};
 
-const getContactById = async (contactId) => {}
+const getContactById = async (contactId) => {
+  try {
+    const contacts = await listContacts();
+    const contactById = contacts.find(({ id }) => id === contactId);
+    return contactById;
+  } catch (err) {
+    console.log(err.message);
+  }
+};
 
-const removeContact = async (contactId) => {}
+const removeContact = async (contactId) => {
+  try {
+    const contacts = await listContacts();
+    const contactToRemove = contacts.find(({ id }) => id === contactId)
+    if (contactToRemove !== undefined) {
+      const { name } = contactToRemove;
+      const newContacts = contacts.filter(({ id }) => id !== contactId);
+      fs.writeFile(contactsPath, JSON.stringify(newContacts, null, 2));
+      return `${name} deleted`
+    }
+    return
+  } catch (err) {
+    console.log(err.message)
+  }
+};
 
-const addContact = async (body) => {}
+const addContact = async (body) => {
+  try {
+    const { name, email, phone } = body;
+    const contacts = await listContacts();
+    const id = Date.now().toString();
```

Comment:
A co jak zarejestruje Ci się 2 użytkowników w tym samym czasie ? W najlepszym wypadku nadpiszą się w najgorszym będą mieli dostęp do swoich danych. Tu są to tylko kontakty ale i tak lepiej jak użył byś biblioteki uuid albo nanoid.